app/src/main/java/com/hedvig/app/feature/embark/passages/externalinsurer/ExternalInsurerFragment.kt

```diff
@@ -3,30 +3,91 @@ package com.hedvig.app.feature.embark.passages.externalinsurer
 import com.hedvig.app.feature.embark.passages.externalinsurer.askforprice.AskForPriceInfoParameter
-import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerBottomSheet
+import com.hedvig.app.feature.embark.passages.previousinsurer.InsurerProviderBottomSheet
+import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerParameter
+import com.hedvig.app.util.extensions.showErrorDialog
 import com.hedvig.app.util.extensions.view.setupInsetsForIme
 import com.hedvig.app.util.whenApiVersion
 import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import org.koin.androidx.viewmodel.ext.android.sharedViewModel
 
 class ExternalInsurerFragment : Fragment(R.layout.previous_or_external_insurer_fragment) {
 
     private val binding by viewBinding(PreviousOrExternalInsurerFragmentBinding::bind)
+
+    private val embarkViewModel: EmbarkViewModel by sharedViewModel()
     private val viewModel: ExternalInsurerViewModel by sharedViewModel()
 
+    private val askForPriceActivityResultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+            if (result.resultCode == RESULT_SKIP) {
+                onContinue()
+            }
+        }
+
     private val insurerData by lazy {
         requireArguments()
             .getParcelable<ExternalInsurerParameter>(DATA)
             ?: throw IllegalArgumentException("No argument passed to ${this.javaClass.name}")
     }
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel.events
+            .flowWithLifecycle(lifecycle)
+            .onEach { event ->
+                if (event is ExternalInsurerViewModel.Event.Continue) {
+                    if (event.providerId == getString(R.string.EXTERNAL_INSURANCE_PROVIDER_OTHER_OPTION)) {
+                        MaterialAlertDialogBuilder(requireContext())
+                            .setTitle(getString(R.string.EXTERNAL_INSURANCE_PROVIDER_ALERT_TITLE))
+                            .setMessage(getString(R.string.EXTERNAL_INSURANCE_PROVIDER_ALERT_MESSAGE))
+                            .setPositiveButton(getString(R.string.ALERT_OK)) { _, _ -> onContinue() }
+                            .show()
+                    } else {
+                        startAskForPrice(event.providerId)
+                    }
+                }
+            }
+            .launchIn(lifecycleScope)
+
+        viewModel.viewState
+            .flowWithLifecycle(lifecycle)
+            .onEach { viewState ->
+                binding.progress.isVisible = viewState.isLoading
+
+                viewState.error?.let {
+                    context?.showErrorDialog(getString(it.getStringRes())) { }
+                }
+
+                viewState.showInsuranceProviders?.let {
```

Comment:
What happens on configuration change here? `onShowInsurers` launches a new bottom sheet, wouldn't we get two bottom sheets in that case?

Seems like this should perhaps be an event 🤔